src/parser/jobs/mch/modules/Reassemble.tsx

```diff
@@ -1,94 +1,128 @@
 	'HEATED_CLEAN_SHOT',
 ]
 
-export default class Reassemble extends Module {
+const SEVERITIES = {
+	BAD_USES: {
+		1: SEVERITY.MINOR,
+		2: SEVERITY.MEDIUM,
+		3: SEVERITY.MAJOR,
+	},
+	DROPPED_USES: {
+		1: SEVERITY.MEDIUM,
+		2: SEVERITY.MAJOR,
+	},
+}
+
+interface ReassembleState {
+	active: boolean
+	lastGcdTime: number
+	gcdHook?: EventHook<Events['action']>
+}
+
+interface ReassembleHistory {
+	badUses: number
+	droppedUses: number
+}
+
+export class Reassemble extends Analyser {
 	static override handle = 'reassemble'
 
-	@dependency private combatants!: Combatants
 	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
-	private badReassembles = 0
-	private droppedReassembles = 0
-	private lastUse = 0
+	private state: ReassembleState = {active: false, lastGcdTime: 0}
+	private history: ReassembleHistory = {badUses: 0, droppedUses: 0}
 
-	private REASSEMBLE_GCDS: number[] = []
-	private OTHER_GCDS: number[] = []
+	private reassembleGcdIds = REASSEMBLE_GCDS.map(key => this.data.actions[key].id)
+	private otherGcdIds = OTHER_GCDS.map(key => this.data.actions[key].id)
 
-	protected override init() {
-		this.REASSEMBLE_GCDS = REASSEMBLE_GCDS.map(key => this.data.actions[key].id)
-		this.OTHER_GCDS = OTHER_GCDS.map(key => this.data.actions[key].id)
+	private gcdFilter = filter<Event>()
+		.source(this.parser.actor.id)
+		.type('action')
+		.action(oneOf([...this.reassembleGcdIds, ...this.otherGcdIds]))
 
-		this.addEventHook('cast', {by: 'player', abilityId: [...this.REASSEMBLE_GCDS, ...this.OTHER_GCDS]}, this.onCast)
-		this.addEventHook('removedebuff', {by: 'player', abilityId: STATUSES.REASSEMBLED.id}, this.onRemoveReassemble)
+	override initialise() {
+		const reassembleFilter = filter<Event>()
+			.source(this.parser.actor.id)
+			.status(this.data.statuses.REASSEMBLED.id)
+
+		this.addEventHook(reassembleFilter.type('statusApply'), this.onReassemble)
+		this.addEventHook(reassembleFilter.type('statusRemove'), this.onRemove)
 		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onCast(event: CastEvent) {
-		if (this.combatants.selected.hasStatus(STATUSES.REASSEMBLED.id)) {
-			this.badReassembles += this.REASSEMBLE_GCDS.includes(event.ability.guid) ? 0 : 1
+	private onReassemble() {
+		if (this.state.active) {
+			this.history.droppedUses += 1
 		}
-		this.lastUse = event.timestamp
+		this.state.gcdHook = this.addEventHook(this.gcdFilter, this.onCast)
+		this.state.active = true
 	}
 
-	private onRemoveReassemble(event: BuffEvent) {
-		if (event.timestamp !== this.lastUse) {
-			this.droppedReassembles++
+	private onCast(event: Events['action']) {
+		if (this.state.active && !this.reassembleGcdIds.includes(event.action)) {
+			this.history.badUses += 1
+		}
+		this.state.lastGcdTime = event.timestamp
+	}
+
+	private onRemove(event: Events['statusRemove']) {
+		if (event.timestamp !== this.state.lastGcdTime) {
+			this.history.droppedUses += 1
+		}
+		if (!(this.state.gcdHook == null)) {
```

Comment:
I think you can just use `if (this.state.gcdHook != null)` here, since the state var is always defined, it's only gcdHook that can be undef, and `!=` against null for undef will still behave as expected. In general, xiva style frowns upon negating falsey checks (but booleans are fine).